core/go/internal/statemgr/domain_context.go

```diff
@@ -249,6 +249,15 @@ func (dc *domainContext) mergeInMemoryMatches(schema components.Schema, states [
 
 }
 
+func (dc *domainContext) GetStates(dbTX *gorm.DB, schemaID tktypes.Bytes32, ids []string) (components.Schema, []*pldapi.State, error) {
+	idsAny := make([]any, len(ids))
+	for i, id := range ids {
+		idsAny[i] = id
+	}
+	query := query.NewQueryBuilder().In(".id", idsAny).Query()
+	return dc.ss.findStates(dc, dbTX, dc.domainName, &dc.contractAddress, schemaID, query, pldapi.StateStatusAll)
```

Comment:
Note: I have not done any merging of unflushed states, so this will only find states that have been flushed to the database. Might need to add "confirmed and available but unflushed" states into the results for completeness.

Reply:
This is done now